hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/InitMetaProcedure.java

```diff
@@ -71,8 +71,8 @@ private static void writeFsLayout(Path rootDir, Configuration conf) throws IOExc
     LOG.info("BOOTSTRAP: creating hbase:meta region");
     FileSystem fs = rootDir.getFileSystem(conf);
     Path tableDir = CommonFSUtils.getTableDir(rootDir, TableName.META_TABLE_NAME);
-    if (fs.exists(tableDir) && !fs.delete(tableDir, true)) {
-      LOG.warn("Can not delete partial created meta table, continue...");
+    if (fs.exists(tableDir)) {
```

Comment:
This is another issue other than the unknown RS issue.
In cluster recreate cloud case, the zk data is not there and so no META region location. So for HM, this is a cluster bootstrap and so as part of that init meta and its FS.  
HBASE-24471 added this extra code of deleting an existing META table FS path.  It is done by considering as a partial created meta table as part of some previous failed bootstrap cc @Apache9 
So this issue is because of the zk data not in new cluster.  Unknown server issue is because of loss of WAL (to be precise the master proc wal)

Reply:
as Zack and I highlighted in the comments above, we found the change was from HBASE-24471 and it's a bit different from that change before (I agreed that change is solving the meta startup/bootstrap problem.). 

let me send another update and add a feature flag to be able to turn off delete partial meta (default is to delete it). Then for the cloud case, we can turn it off for the cloud use case that ZK data has been deleted before we come up a better way to tell what is partial. (sorry that I don't have a good way to validate if meta is `partial` from previous bootstrap.  any suggestion?)

Reply:
I agree that in cloud cluster recreate case, this META FS delete is a bigger concern.  At least the other case of unknown servers, we can make the table to be disabled at least.. But here not such workaround even possible I fear. Thanks for pointing out.
Now taking a step back.  All these decision points in HM start regarding bootstrapping or AM work is based on the assumption that HBase is a continuously running cluster.  So the cluster comes up first and then data getting persisted in FS over the run.  Now in cloud there is a very useful feature of delete a cluster and keep the data in blob store and later when needed, recreate the cluster pointing to the existing data.  Many of the decision making in HM start is not holding right at that time! (Like the one above which thinks that if there is no META location in zk, means meta table was never been online and no data in it)  So what we need is a way to know that whether this cluster start is from an existing data (cluster recreate)..  All these decisions can be based on that check result (?)  Even the unknown server handling also so that it will happen ONLY in this special cloud case and only once.